packages/eslint-plugin-tslint/src/rules/config.ts

```diff
@@ -1,10 +1,27 @@
 import { ESLintUtils } from '@typescript-eslint/utils';
-import memoize from 'lodash/memoize';
 import type { RuleSeverity } from 'tslint';
 import { Configuration } from 'tslint';
 
 import { CustomLinter } from '../custom-linter';
 
+function memoize<T extends (...args: never[]) => unknown>(
+  func: T,
+  resolver: (...args: Parameters<T>) => string,
+): T {
+  const cache = new Map<string, ReturnType<T>>();
+  const memoized = function (...args) {
+    const key = resolver(...(args as Parameters<T>));
+
+    if (cache.has(key)) {
+      return cache.get(key)!;
+    }
+    const result = func(...args);
+    cache.set(key, result as ReturnType<T>);
+    return result;
+  } as T;
+  return memoized;
+}
+
```

Comment:
can we extract this change to standalone PR?

Reply:
Yeah will do. I was just looking to see if we had a lodash dependency (because the rule tester needed one) and I noticed this plugin used the entire lodash package for just one function!

I'll extract it out because it's not worth the massive dependency, esp given we can simplify the function massively.

Reply:
if possible we should just drop it

Reply:
https://github.com/typescript-eslint/typescript-eslint/pull/6810